src/adagrams.js

```diff
@@ -1,15 +1,181 @@
+const LETTER_POOL = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+};
+
+const getRandomIdx = () => {
+  const NUM_OF_LETTERS_IN_LETTER_BANK = 96
+  return Math.floor(Math.random() * NUM_OF_LETTERS_IN_LETTER_BANK)
+};
+
+const getLetterBank = (LETTER_POOL)=> {
+  let letterBank = []
+  for(let letter in LETTER_POOL){
+    const freq = LETTER_POOL[letter]
+	    letterBank.push(...letter.repeat(freq).split(''))
+  }
+  return letterBank
+};
+const setFreqOfLettersInHand = (hand) => {
+  const freqOfLettersInHand = {};
+  for(let ch of hand){
+    if(Object.hasOwn(hand, ch)){
+      freqOfLettersInHand[ch]+=1;
+    }else{
+      freqOfLettersInHand[ch]=1 ; 
+    }
+  };
+  return freqOfLettersInHand;
+};
+
+const getFreqOfLetter = (letter, hand) => {
+  let freq = 0
+  for(let char of hand){
+    if(char===letter){
+      freq+=1
```

Comment:
Nitpick: Spacing, using a linter can help you identify style issues so you can fix them up during the development process. That way your code is neat and ready for review when you open a PR. Typically, in industry, you'll be asked to fix up your code so that it follows style guidelines before someone will review it.

There are several more places in this PR that have spacing issues that need to be fixed up but I'll leave them so I don't clutter this review with more spacing comments.

```suggestion
    if (char ===l etter) {
      freq += 1
```